File: Classes/BEMCircle.m
//
//  BEMCircle.m
//  SimpleLineGraph
//
//  Created by Bobo on 12/27/13.
//  Copyright (c) 2013 Boris Emorine. All rights reserved.
//

#import "BEMCircle.h"

@implementation BEMCircle

- (id)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        // Initialization code
        self.backgroundColor = [UIColor clearColor];
    }
    return self;
}

- (void)drawRect:(CGRect)rect
{
    CGContextRef ctx = UIGraphicsGetCurrentContext();
    CGContextAddEllipseInRect(ctx, rect);
    [[UIColor whiteColor] set];
    CGContextFillPath(ctx);
}

@end

File: Swift Implementation/BEMCircle.swift
//
//  BEMCircle.swift
//  SimpleLineGraph
//
//  Created by Sam Spencer on 7/30/17.
//  Copyright © 2017 Samuel Spencer. All rights reserved.
//

import UIKit
import Foundation
import CoreGraphics

/// Class to draw the circle for the points (Swift).
class BEMCircle_Swift: UIView {

    /// Set to YES if the data point circles should be constantly displayed. NO if they should only appear when relevant.
    public var shouldDisplayConstantly: Bool = false
    
    /// The point color
    public var color: UIColor = .white
    
    /// The value of the point
    public var absoluteValue: CGFloat = 0.0

    override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = .clear
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.backgroundColor = .clear
    }
    
     override func draw(_ rect: CGRect) {
        // Drawing code
        let context = UIGraphicsGetCurrentContext()
        context?.addEllipse(in: rect)
        color.set()
        context?.fillPath()
     }
}
